evap/staff/importers/json.py

```diff
@@ -0,0 +1,445 @@
+        programs = [
+            self._get_program(c["cprid"]) for c in data["courses"] if c["cprid"] not in settings.IGNORE_PROGRAMS
+        ]
+        course.programs.add(*programs)
+
+    def _import_course_from_unused_exam(self, data: ImportEvent) -> Course | None:
+        split_title = data["title"].split(":", 1)
+        if len(split_title) < 2:
+            return None
+        prefix = split_title[0].strip()
+
+        try:
+            course_type = CourseType.objects.get(import_names__contains=[prefix])
+        except CourseType.DoesNotExist:
+            return None
+
+        data["title"] = split_title[1].strip()
+        data["title_en"] = data["title_en"].split(":", 1)[1].strip() if ":" in data["title_en"] else data["title_en"]
+
+        return self._import_course(data, course_type)
+
+    # pylint: disable=too-many-locals
+    def _import_evaluation(self, course: Course, data: ImportEvent) -> Evaluation:
+        if "appointments" not in data:
+            self.statistics.warnings.append(
+                WarningMessage(obj=course.name, message="No dates defined, using default end date")
+            )
+            course_end = datetime.combine(self.default_course_end, time())
+        else:
+            last_appointment = max(data["appointments"], key=lambda x: x["end"])
+            course_end = datetime.strptime(last_appointment["end"], self.DATETIME_FORMAT)
+
+        if data["isexam"]:
+            # Set evaluation time frame of three days for exam evaluations:
+            evaluation_start_datetime = course_end.replace(hour=8, minute=0, second=0, microsecond=0) + timedelta(
+                days=1
+            )
+            evaluation_end_date = (course_end + timedelta(days=3)).date()
+
+            name_de = data["title"].split(" - ")[-1] if " - " in data["title"] else "Prüfung"
+            name_en = data["title_en"].split(" - ")[-1] if " - " in data["title_en"] else "Exam"
+
+            weight = 1
+
+            # If events are graded for any program, wait for grade upload before publishing
+            wait_for_grade_upload_before_publishing = any(grade["scale"] for grade in data["courses"])
+            # Update previously created main evaluation
+            course.evaluations.all().update(
+                wait_for_grade_upload_before_publishing=wait_for_grade_upload_before_publishing
+            )
+        else:
+            # Set evaluation time frame of two weeks for normal evaluations:
+            # Start datetime is at 8:00 am on the monday in the week before the event ends
+            evaluation_start_datetime = course_end.replace(hour=8, minute=0, second=0, microsecond=0) - timedelta(
+                weeks=1, days=course_end.weekday()
+            )
+            # End date is on the sunday in the week the event ends
+            evaluation_end_date = (course_end + timedelta(days=6 - course_end.weekday())).date()
+
+            name_de, name_en = "", ""
+
+            weight = 9
```

Comment:
I general, it's not super awesome that the knowledge about weights and timedeltas for exam evaluations are duplicated here; okay to fix in a follow up